File: practice/algorithms/warmup/compare_the_triplets/CompareTheTriplets.scala
object CompareTheTriplets extends App {
  val lines = Source.stdin.getLines().toList
  val alicesRating = lines(0).split(" ").map(_.toInt)
  val bobsRating = lines(1).split(" ").map(_.toInt)

  val differentRatings = alicesRating.zip(bobsRating).filter(pair => pair._1 != pair._2)
  val aliceScore = differentRatings.count(pair => pair._1 > pair._2)
  val bobScore = differentRatings.length - aliceScore

  println(aliceScore + " " + bobScore)
}

File: practice/algorithms/warmup/compare_the_triplets/compare_the_triplets.rb
a0, a1, a2 = gets.strip.split(' ')
a0 = a0.to_i
a1 = a1.to_i
a2 = a2.to_i
b0, b1, b2 = gets.strip.split(' ')
b0 = b0.to_i
b1 = b1.to_i
b2 = b2.to_i

alice_score = 0
bob_score = 0

alice_score = alice_score + 1 if a0 > b0
alice_score = alice_score + 1 if a1 > b1
alice_score = alice_score + 1 if a2 > b2

bob_score = bob_score + 1 if a0 < b0
bob_score = bob_score + 1 if a1 < b1
bob_score = bob_score + 1 if a2 < b2

puts "#{alice_score} #{bob_score}"